server/controller/user.controller.js

```diff
@@ -35,274 +34,233 @@ import { User } from '../database/models';
 
 const userController = express.Router();
 
-function createUser(email, password) {
+/**
+ * Create a User with email, password, is_admin default = FALSE
+ */
+userController.createUser = (email, password) => {
   const data = {
     email,
     hashedPassword: generateHashedPassword(password),
-    avatarUrl: '',
-    avatarType: '',
-    firstName: '',
-    lastName: '',
-    bio: '',
-    isAdmin: false,
-    coursesTaken: [],
-    gradDate: { year: undefined, term: '' },
-    major: '',
-    minor: '',
-    catalogYear: undefined,
-    degreePlanId: undefined,
   };
 
-  user = new User(data);
-  return user.save();
-}
+  return new User(data).save();
+};
 
 /**
  * POST/
  * Register a user
  */
 userController.post('/register', validateRegisterUser, async (req, res) => {
-  const errorsAfterValidation = validationResult(req);
-  if (!errorsAfterValidation.isEmpty()) {
-    res.status(400).json({
-      code: 400,
-      errors: errorsAfterValidation.mapped(),
-    });
-  } else {
+  validationHandler(req, res, async () => {
     try {
       const { email, password } = req.body;
 
-      const user = await User.findOne({ email });
```

Comment:
Not a problem to not call await? Is this because you are defining an async callback function in the .then()?

Reply:
Inoticed you call await in the POST API for login but call an async function in this one.

Reply:
Yes, I just remove the await in login/ 
The reason is that I put `.then()` after findOne so we don't need to call await.